tracer/src/Datadog.Trace.Tools.Runner/Utils.cs

```diff
@@ -280,5 +280,53 @@ public static string GetEnvironmentVariable(string key, string defaultValue = nu
 
             return defaultValue;
         }
+
+        public static bool CheckAgentConnection(string agentUrl)
+        {
+            var env = new Dictionary<string, string>();
+            if (!string.IsNullOrWhiteSpace(agentUrl))
+            {
+                env["DD_TRACE_AGENT_URL"] = agentUrl;
+            }
+
+            var tracerSettings = new Configuration.TracerSettings(new DictionaryConfigurationSource(env));
+
+            using (var tcpClient = new System.Net.Sockets.TcpClient())
+            {
+                try
+                {
+                    tcpClient.Connect(tracerSettings.AgentUri.Host, tracerSettings.AgentUri.Port);
+                    Console.WriteLine($"Connection with the Datadog Agent on {tracerSettings.AgentUri} was successful.");
+                }
+                catch (Exception ex)
+                {
+                    Console.WriteLine($"Error connecting to the Datadog Agent on {tracerSettings.AgentUri}.");
+                    Console.WriteLine(ex);
+                    return false;
+                }
+            }
+
+            return true;
+        }
```

Comment:
Given this functionality is already baked into the `AgentWriter`, couldn't replace all this with something like

```csharp
var tracerSettings = TracerSettings.FromDefaultSources();
var agentWriter = CIAgentWriter(TracerSettings settings)
var canConnect = await agentWriter.Ping();
await agentWriter.FlushAndClose();
```

Reply:
...and after all this time, I didn't realize we had a `Ping` api...

Reply:
done in 1f293bb